d2core/d2asset/composite.go

```diff
@@ -126,8 +127,12 @@ func (c *Composite) createMode(animationMode, weaponClass string, direction int)
 		return nil, err
 	}
 
-	if direction >= cof.NumberOfDirections {
-		return nil, errors.New("invalid direction")
+	// oh god how do i math
+	offset := (64 / cof.NumberOfDirections) / 2
+	entityDirection := int(math.Trunc((float64(direction+offset)-64.0)*(-float64(cof.NumberOfDirections)/-64.0) + float64(cof.NumberOfDirections)))
```

Comment:
It looks like `math.Trunc` is redundant
```suggestion
	int((float64(direction+offset)-64.0)*(-float64(cof.NumberOfDirections)/-64.0) + float64(cof.NumberOfDirections))
```